### ISSUE TITLE
Container with Alpine Linux can't connect to the local Edge Hub on EFLOW

### ISSUE THREAD

**Issue by iconics-pavels** [Azure/iotedge-eflow#160]
**Description**
We have a sample container using the Alpine Linux 3.14.6. IoT Edge is running on EFLOW using the Linux Mariner 1.0.20220504. Sample application is .NET Core 3.1 but it's reproducible also with .NET 6.0. If we specify following container create options for our container then it can't connect to the local Edge Hub. We need to specify following create options in our real container (e.g. to discover the BACnet network). 

{
    "NetworkingConfig": {
        "EndpointsConfig": {
            "host": {}
        }
    },
    "HostConfig": {
        "NetworkMode": "host"
    }
}

Everything works fine when the HOST setting is not specified. Also when we create our container using an Ubuntu base image then it works with the HOST setting. So it's a combination of the Alpine Linux, EFLOW and the HOST setting.

**To Reproduce**
1. Prepare a device in the IoT Hub.
2. Install and configure EFLOW.
3. Deploy our test container including the HOST setting (icodevtest/iotedgetestmodule:amd64).
4. Check the behavior (e.g. via iotedge logs). Container can't connect to the local Edge Hub.

**Windows Host OS (please complete the following information):**
 - EFLOW 1.0.20220504 running on Windows Server 2022 Datacenter 21H2 (20348.469) x64
 - Container is based on Alpine Linux 3.14.6 (mcr.microsoft.com/dotnet/aspnet:3.1-alpine3.14)

**Sample code**
[IoTEdgeTestModule.zip](https://github.com/Azure/iotedge-eflow/files/8946707/IoTEdgeTestModule.zip)

**IoT Edge check summary**
[iotedge_check.txt](https://github.com/Azure/iotedge-eflow/files/8946867/iotedge_check.txt)

**Comment by iconics-pavels** [comment]
Adding some more details after we did the initial investigation during the meeting.

We checked logs of the tcpdump in the edgeHub module and we didn't see any communication between the edgeHub and our test module when the HOST setting was set. When we removed it from the create options of our module then we saw a communication as expected.

We also tried to allow all communication through the firewall on the EFLOW machine, but it also didn't help.

**Comment by iconics-rich** [comment]
[support_bundle.zip](https://github.com/Azure/iotedge-eflow/files/8977212/support_bundle.zip)

Adding support bundle from system where issue occurs.

**Comment by fcabrera23** [comment]
We were able to narrow down the cause of this issue. To summarize what we found:

1)	In our test module we observed the following environment variables set:
HOSTNAME = W2022QA2-EFLOW
IOTEDGE_GATEWAYHOSTNAME = w2022qa2-eflow
2)	We noticed that the case of these two values did not match. The actual hostname of the VM is the all-caps version
3)	We stepped inside our test container and saw we could ping W2022QA2-EFLOW, but NOT w2022qa2-eflow
4)	After manually updating /etc/hosts file in the container to also have the lowercase version of the hostname, the connection from our module to the edgeHub module succeeded!
